### PR TITLE
[10.0][IMP]performance account_analytic_parent

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**
solves issue #198 

Thanks @pedrobaeza  and @mreficent for the help!

### CODE REVIEW

**Comment by mreficent** [comment]
:red_circle:

**Review by MiquelRForgeFlow — APPROVED** [review]
LGTM

File: account_analytic_parent/models/account_analytic_account.py [hunk 1]
```diff
@@ -27,10 +27,25 @@ def _compute_debit_credit_balance(self):
...
         """
         super(AccountAnalyticAccount, self)._compute_debit_credit_balance()
-        for account in self:
-            account.debit += sum(account.mapped('child_ids.debit'))
-            account.credit += sum(account.mapped('child_ids.credit'))
-            account.balance += sum(account.mapped('child_ids.balance'))
+        analytic_line_obj = self.env['account.analytic.line']
+        # compute only analytic line
+        for account in self.filtered(lambda x: x.child_ids):
+            domain = [('account_id', 'in', account.child_ids.ids)]
```
Comment (pedrobaeza) [inline account_analytic_parent/models/account_analytic_account.py]: operator should be `child_of` for including account itself.

Comment (pedrobaeza) [reply account_analytic_parent/models/account_analytic_account.py]: and do the test over `self.ids`, not `self.child_ids.ids`

Comment (AaronHForgeFlow) [reply account_analytic_parent/models/account_analytic_account.py]: account itself is not included in the loop but it was already included when calling super, then in the loop we increment the values by adding the children ones. But maybe is less confusing doing that way, it's ok.

Comment (AaronHForgeFlow) [reply account_analytic_parent/models/account_analytic_account.py]: This is resolved, isn't it?

Comment (AaronHForgeFlow) [reply account_analytic_parent/models/account_analytic_account.py]: @pedrobaeza Can you please update your review? IMO your comment is attended.

File: account_analytic_parent/models/account_analytic_account.py [hunk 2]
```diff
@@ -27,10 +27,25 @@ def _compute_debit_credit_balance(self):
...
-        for account in self:
-            account.debit += sum(account.mapped('child_ids.debit'))
-            account.credit += sum(account.mapped('child_ids.credit'))
-            account.balance += sum(account.mapped('child_ids.balance'))
+        analytic_line_obj = self.env['account.analytic.line']
+        # compute only analytic line
+        for account in self.filtered(lambda x: x.child_ids):
+            domain = [('account_id', 'child_of', account.ids)]
+            credit_groups = analytic_line_obj.read_group(
+                domain=domain + [('amount', '>=', 0.0)],
```
Comment (pedrobaeza) [inline account_analytic_parent/models/account_analytic_account.py]: No need to include 0 results

Comment (pedrobaeza) [reply account_analytic_parent/models/account_analytic_account.py]: I mean they are not going to add anything to the sum, as their amount is 0, so better to strip them.

Comment (AaronHForgeFlow) [reply account_analytic_parent/models/account_analytic_account.py]: yes, it makes sense

**Pull request merged**